lab_1/main.py

```diff
@@ -98,7 +213,30 @@ def detect_language_advanced(unknown_profile: dict, profiles: list, languages: l
     :param top_n: a number of the most common words
     :return: a language
     """
-    pass
+    if (not isinstance(unknown_profile, dict)
+            or not isinstance(profiles, list)
+            or not isinstance(languages, list)
+            or not isinstance(top_n, int)):
+        return None
+    # sort the reports by score
+    reports = []
+    for profile in profiles:
+        if profile["name"] in languages or not languages:
+            report = compare_profiles_advanced(unknown_profile, profile, top_n)
+            reports.append(report)
+    reports = sorted(reports, key=lambda x: x["score"], reverse=True)
+    # sort in alphabetically order if some languages have the same max scores
+    if len(reports) == 0:
+        return None
+    if len(reports) > 1:
+        if reports[0]["score"] == reports[1]["score"]:
+            reports_with_max_score = []
+            for element in reports:
+                if element["score"] == reports[0]["score"]:
+                    reports_with_max_score.append(element)
+            reports = sorted(reports_with_max_score, key=lambda x: x["name"])
```

Comment:
this function is hard to read
Maybe is there a more understandable way?

Reply:
I've made some changes in the block of function, where we need to sort the name of languages in alphabetically order if the languages have the same max scores. I hope, the function has become more readable and understandable than earlier.

Reply:
it's better now, yep